### ISSUE TITLE
Adding content at runtime when it reached the end

### ISSUE THREAD

**Issue by DannyBen** [richhollis/vticker#17]
Trying to figure it out from the code, but not sure if it was designed for it or not.

I am trying to use it as an RSS ticker, so ideally:

1. I would need to know when the ticker has reached its end, so I can refetch a fresh copy of the RSS feed
2. I would need to remove all `li` elements, and re-add the new ones.

Seeing [this line in the init method](https://github.com/richhollis/vticker/blob/master/jquery.vticker.js#L107) got me worried, since if I am reading this correctly, the item count is calculated only on init.

Is what I am trying to do feasible given the current design?

**Comment by richhollis** [comment]
Hi Danny. 

Perfectly possible to add some additional events (or extend existing ones) so you know when you've reached the end of the list. Maybe even add a parameter for controlling looping.

Yeah, currently the list size is calculated during init. So for your needs it might just be easier to roll a new ticker when you reach the end of the list. So when you reach list end you could:

- Stop the existing ticker
- Create a new absolutely positioned list (UL) element with your new items - in the same place as the other
- Attach vticker to new element
- Transition between old and new lists (crossfade etc.) or just a simple hide/show
- Call the "remove" method to remove the original ticker
- Repeat

Thoughts?

Cheers
Rich

**Comment by DannyBen** [comment]
Thanks for the detailed reply Rich.

I was hoping for a little more straight forward way. From my initial tests, it seems like maybe I found a way to deal with at least part of the problem here.

I noticed that items are simply removed from the top and appended to the bottom.
So - when I fetch more items, I append them to the list, and after that I call  `$('#feed').vTicker` again, which seems to do the right things like make all elements the same height etc.

This is a simplified version of the coffeesript:

```coffeescript
parseRSS = (url, callback) ->
  # truncated for brevity

addFeedEntries = (feed) ->
  for entry in feed.entries
    $('#feed ul').append "<li>#{entry.title}</li>"
  $('#feed').vTicker pause: 8000

loadFeed = (url) ->
  parseRSS url, addFeedEntries

$ ->
  loadFeed 'http://somefeed.com/rss.xml'
  loadFeed 'http://anotherfeed.com/rss.xml'
```

If the above makes sense, then the only issue left for me to try and handle is when exactly to I execute a refresh of the feed (which means call `loadFeed` again, and remove the old `li` items).

**Comment by DannyBen** [comment]
Here is my [attempt on jsbin](https://jsbin.com/tuleliq/edit?js,console,output)

These are the features I am trying to obtain:
1. Load one or more RSS feed
2. Re-fetch the feed when reaching the end of the queue
3. Make the transition between the end of the queue to the beginning of the new queue as natural as possible

**Comment by richhollis** [comment]
Hi Danny

I came across you on a Rails config gem issue yesterday rather randomly! Spooky.

Your jsbin is looking pretty good. So what's left? Just the smooth transition? You're probably going to need to load the feed much sooner though than the end - maybe when you're halfway through?

**Comment by DannyBen** [comment]
Richard - are you saying I talk too much... ? :smile: 

I think actually not much is left - I am quite happy with how it works now. The 'on tick' event is sufficient and generic enough so anyone can do with it as they please.

Perhaps one option would be nice to have, to say to vticker "dont re-append my items when you are done with them". This will just save me the need to "delete all but the currently visible item" and will allow me to indeed, as you say, go and re-fetch new items much sooner.

But other than this small suggestion, can close this ticket as far as I am concerned, with my thanks.

**Comment by richhollis** [comment]
Haha! Yes - way too much! ;)

Cool. Sounds good. So sounds like an additional option "loop" or something would be ideal for your needs? When loop is false it won't re-append the items.

**Comment by DannyBen** [comment]
This would be nice, yes (assuming of course you agree it might be useful for other people as well).

**Comment by richhollis** [comment]
Hi Danny

Can you take a look at the changes I made (alwaysAppendItems) and let me know if it works for your use case.

Does your user have any explicit prev/next buttons to control the sequence? If you do then this approach won't work. Pause/hover is OK though.

Thanks!
Rich

**Comment by DannyBen** [comment]
Rich - seems to be working perfectly. I do not use any prev/next controls.

This is what I did, to confirm:
A: Replaced the content of my vticker.js file with [this](https://github.com/richhollis/vticker/blob/master/js/jquery.vticker.js)

B. Changed my initializer from:
```coffee
$('#feed').vTicker pause: pause, speed: speed
```

to
```coffee
$('#feed').vTicker pause: pause, speed: speed, alwaysAppendItems: false
```

C. Removed the line I had previously that was responsible for "deleting everything on loop"

```coffee
$('#feed ul li').not(':first').remove()
```

The result seems to be as expected. Items are not appended, and since I am re-fetching RSS items when it reaches the end, and append them myself, it all still behaves "loopy"

One small suggestion - perhaps the parameter name is too verbose - why not just `autoLoop` or `autoAppend` ? the words "always" and "items" are a little redundant I think.

Thanks for implementing.
